OpenRA.Mods.Common/Traits/Tooltip.cs

```diff
@@ -13,11 +13,11 @@
 
 namespace OpenRA.Mods.Common.Traits
 {
-	public abstract class TooltipInfoBase : ITraitInfo
+	public abstract class TooltipInfoBase : UpgradableTraitInfo
 	{
 		[Translate] public readonly string Name = "";
 
-		public abstract object Create(ActorInitializer init);
+		public override object Create(ActorInitializer init) { return this; }
```

Comment:
Returning `this` here is bogus (that is a traitinfo, not a trait).  You should be able to remove this line completely.